proj-9.3.0-r1 fails to compile with gcc-15.0.0_pre20240804

Two files are missing include of cstdint and fail to compile with gcc-15.0.0_pre20240804
Reported downstream as https://bugs.gentoo.org/937472

#### Example of problem
````
FAILED: src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DPROJ_DATA=\"/usr/share/proj\" -DPROJ_HAS_PTHREADS -I/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src -I/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/include -I/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0_build/src  -march=native -O3 -flto=jobserver -mprefer-vector-width=512 -fno-vect-cost-model -pipe -std=c++11 -fvisibility=hidden -Wall -Wextra -Wswitch -Wshadow -Wunused-parameter -Wmissing-declarations -Wformat -Wformat-security -MD -MT src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o -MF src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o.d -o src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o -c /var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp: In function ‘void process(FILE*)’:
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:114:45: error: ‘uint8_t’ does not name a type
  114 |         if (nLineNumber == 1 && static_cast<uint8_t>(s[0]) == 0xEF &&
      |                                             ^~~~~~~
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:56:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   55 | #include "utils.h"
  +++ |+#include <cstdint>
   56 | // clang-format on
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:115:25: error: ‘uint8_t’ does not name a type
  115 |             static_cast<uint8_t>(s[1]) == 0xBB &&
      |                         ^~~~~~~
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:115:25: note: ‘uint8_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:116:25: error: ‘uint8_t’ does not name a type

.
.
.
.

FAILED: src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DPROJ_DATA=\"/usr/share/proj\" -DPROJ_HAS_PTHREADS -I/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src -I/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/include -I/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0_build/src  -march=native -O3 -flto=jobserver -mprefer-vector-width=512 -fno-vect-cost-model -pipe -std=c++11 -fvisibility=hidden -Wall -Wextra -Wswitch -Wshadow -Wunused-parameter -Wmissing-declarations -Wformat -Wformat-security -MD -MT src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o -MF src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o.d -o src/apps/CMakeFiles/cs2cs.dir/cs2cs.cpp.o -c /var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp: In function ‘void process(FILE*)’:
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:114:45: error: ‘uint8_t’ does not name a type
  114 |         if (nLineNumber == 1 && static_cast<uint8_t>(s[0]) == 0xEF &&
      |                                             ^~~~~~~
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:56:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   55 | #include "utils.h"
  +++ |+#include <cstdint>
   56 | // clang-format on
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:115:25: error: ‘uint8_t’ does not name a type
  115 |             static_cast<uint8_t>(s[1]) == 0xBB &&
      |                         ^~~~~~~
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:115:25: note: ‘uint8_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/var/tmp/portage/sci-libs/proj-9.3.0-r1/work/proj-9.3.0/src/apps/cs2cs.cpp:116:25: error: ‘uint8_t’ does not name a type
  116 |             static_cast<uint8_t>(s[2]) == 0xBF) {
      |                         ^~~~~~~
''''





#### Environment Information

- PROJ version proj-9.3.0-r1
- Operation System Information
- gcc version 15.0.0 20240804 (experimental) (Gentoo 15.0.0_pre20240804 p8)


#### Installation method

- from source

#### Patch files

[cct.cpp-gcc15-fix.patch](https://github.com/user-attachments/files/16518246/cct.cpp-gcc15-fix.patch)
[cs2cs.cpp-gcc15-fix.patch](https://github.com/user-attachments/files/16518247/cs2cs.cpp-gcc15-fix.patch)

--- a/src/apps/cct.cpp
+++ b/src/apps/cct.cpp
@@ -79,6 +79,7 @@ Thomas Knudsen, thokn@sdfe.dk, 2016-05-25/2017-10-26
 #include <string.h>
 
 #include <algorithm>
+#include <cstdint>
 #include <fstream> // std::ifstream
 #include <iostream>
 

--- a/src/apps/cs2cs.cpp
+++ b/src/apps/cs2cs.cpp
@@ -36,6 +36,7 @@
 #include <string.h>
 
 #include <cassert>
+#include <cstdint>
 #include <iostream>
 #include <string>
 #include <vector>

